packages/insomnia-app/app/ui/components/wrapper-home.js

```diff
@@ -161,32 +163,55 @@ class WrapperHome extends React.PureComponent<Props, State> {
 
         const core = Math.random() + '';
 
-        // Create in-memory filesystem to perform clone
-        const plugins = git.cores.create(core);
-        const fsPlugin = MemPlugin.createPlugin();
-        plugins.set('fs', fsPlugin);
+        let fsClient = MemClient.createClient();
+
+        const originalUrl = translateSSHtoHTTP(repoSettingsPatch.uri);
 
         // Pull settings returned from dialog and shallow-clone the repo
-        const { credentials, uri: url } = repoSettingsPatch;
+        const cloneParams = {
+          ...gitCallbacks({
+            username: repoSettingsPatch.credentials.username,
+            password: repoSettingsPatch.credentials.token,
+          }),
+          fs: fsClient,
+          http: httpClient,
+          core,
+          dir: GIT_CLONE_DIR,
+          gitdir: GIT_INTERNAL_DIR,
+          singleBranch: true,
+          url: originalUrl,
+          depth: 1,
+        };
         try {
-          await git.clone({
-            core,
-            dir: GIT_CLONE_DIR,
-            gitdir: GIT_INTERNAL_DIR,
-            singleBranch: true,
-            url,
-            ...credentials,
-            depth: 1,
-            noGitSuffix: true,
-          });
-        } catch (err) {
-          showError({ title: 'Error Cloning Repository', message: err.message, error: err });
-          return false;
+          await git.clone(cloneParams);
+        } catch (originalUrlError) {
+          if (cloneParams.url.endsWith('.git')) {
+            showError({ title: 'Error Cloning Repository', message: originalUrlError.message });
+            return;
+          }
+
+          const dotGitUrl = addDotGit(cloneParams);
+          try {
+            fsClient = MemClient.createClient();
+            await git.clone({
+              ...cloneParams,
+              fs: fsClient,
+              url: dotGitUrl,
+            });
+
+            // by this point the clone was successful, so update with this syntax
+            repoSettingsPatch.uri = dotGitUrl;
+          } catch (dotGitError) {
+            showError({
+              title: 'Error Cloning Repository: failed to clone with and without `.git` suffix',
+              message: `failed to clone with original url (${originalUrl}): ${originalUrlError.message};\n\nalso failed to clone with \`.git\` suffix added (${dotGitUrl}): ${dotGitError.message}`,
```

Comment:
Just to follow the other error messaging patterns in the app, the modal title is usually quite brief and the message contains all the content.

```suggestion
              title: 'Error Cloning Repository',
              message: `Failed to clone with original url (${originalUrl}): ${originalUrlError.message};\n\nAlso failed to clone with \`.git\` suffix added (${dotGitUrl}): ${dotGitError.message}`,
```

Reply:
because the title is what @nijikokun asked for, I'm going to leave it https://github.com/Kong/insomnia/pull/3111#discussion_r599160786

I've updated the capitalization in 6e42a57a310116085b3dedc866486940aba93a4f